### ISSUE TITLE
[Feature Request]: RunInference: investigate adding optional batching flag

### ISSUE THREAD

**Issue by yeandy** [apache/beam#21863]
### What would you like to happen?

Look into adding a flag that users can specify to turn off `BatchElements`.

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-py-core

**Comment by TheNeuralBit** [comment]
Note that the ability to elide batching/unbatching is the primary value provided by https://s.apache.org/batched-dofns right now.

**Comment by damccorm** [comment]
@yeandy do we actually need this or is setting the max batch size to 1 enough? Is the idea here just to offer a convenience flag?

**Comment by yeandy** [comment]
Not quite. I think the original idea was that users may want to pass in pre-batched inputs (i.e. tensors that include a batched dimension). If they did that, however, then an additional dimension would be added because we call `torch.stack()` or `np.stack()` inside `run_inference()`. 

The thinking was that we could pass in some sort of flag to skip `BatchElements()` (and also `torch.stack()` or `np.stack()`) to allow users to more easily operate with this type of batched data.

The Batched DoFns from Brian's work should enable us to do this in better way.

**Comment by damccorm** [comment]
> The Batched DoFns from Brian's work should enable us to do this in better way.

For future reference/any larger audience, hooking into this is tracked in #21440

**Comment by damccorm** [comment]
Once this is done, we should update [run_inference_multi_model.ipynb](https://github.com/apache/beam/pull/24437/files/d29825911841635ba03c7d2d80943d0326d149de#diff-280d6ee1fd5f85ade179b3d50cb0914f26155a5865dd9dddb113d751e5f4a888) to remove the workarounds we're using there
